velox/type/Type.h

```diff
@@ -15,6 +15,7 @@
  */
 #pragma once
 
+#include <boost/algorithm/string.hpp>
```

Comment:
We should avoid this header in Type.h to improve build time.
`isDecimalName` does not have to be inlined I supposed. It does not get called in the execution path.